### PR TITLE
Added page to build list.

### PR DESCRIPTION
**Pull request by Raffo**
This adds to the drone-go the capability to use the build list pagination.

### CODE REVIEW

**Comment by tboerger** [comment]
Maybe it's better to add a listing struct that could accept more attributes in the future, like per page.

**Comment by tboerger** [comment]
As an example https://github.com/hetznercloud/hcloud-go/blob/master/hcloud/image.go with ImageListOpts

**Comment by Raffo** [comment]
Maybe, but that doesn't sound like a required change right now, the function is taking just 3 parameters. Do you require this change to get this merged?

**Comment by tboerger** [comment]
I don't know what @bradrydzewski thinks about it. I personally would prefer that.

**Comment by Raffo** [comment]
Ok, then I'll ask @bradrydzewski to give his opinion, it doesn't cost much to me to make the change.

**Comment by Raffo** [comment]
@bradrydzewski do you mind giving your opinion on this such that we can go on integrating this change?

**Comment by bradrydzewski** [comment]
lets take a similar approach to this:
https://github.com/drone/go-scm/blob/master/scm/driver/github/repo.go#L81

**Comment by tboerger** [comment]
Yeah, more or less the same as I suggested

**Comment by Raffo** [comment]
Ok, perfect, I will implement that later and update the PR, thanks @bradrydzewski @tboerger for your support.

**Comment by Raffo** [comment]
@tboerger @bradrydzewski I made the changes you suggested, let me know your feedback.

File: drone/client.go [hunk 1]
```diff
@@ -218,9 +230,9 @@ func (c *client) BuildLast(owner, name, branch string) (*Build, error) {
 
 // BuildList returns a list of recent builds for the
 // the specified repository.
-func (c *client) BuildList(owner, name string) ([]*Build, error) {
+func (c *client) BuildList(owner, name string, opts *ListOptions) ([]*Build, error) {
```
Comment (bradrydzewski) [inline drone/client.go]: the only suggestion I have is to remove pointer to reduce possibility of a nil pointer when accessing `opts.Page`

```
-opts *ListOptions
+opts ListOptions
```

Comment (Raffo) [reply drone/client.go]: Thanks for your review, I addressed the comment.

**Pull request merged**